pkg/utils/logutils.go

```diff
@@ -25,5 +25,10 @@ func InitializeLog(isDebug bool) {
 
 // Logger returns the configured Logrus logger instance.
 func Logger() *logrus.Logger {
+	if logger == nil {
```

Comment:
I don't think it's a clean solution, why we have an init for this object if we do it here anyway?
If that's an object that has an init all the other objects using it should hold an instance and call the init. 
Worth thinking about removing the logs from the code and keeping a single logger in main to print steps/errors.

Reply:
What is the point with that?
We want to it to print the debugs one on debug mode not sure about the removing

Reply:
We are working with objects, so every object using logger should create an instance of logger, logger initing itself is not the correct way to work with objects.